palette/support/src/dispatch.rs

```diff
@@ -861,7 +862,17 @@ macro_rules! decl_module {
 			$crate::sr_primitives::traits::OnInitialize<$trait_instance::BlockNumber>
 			for $module<$trait_instance$(, $instance)?> where $( $other_where_bounds )*
 		{
-			fn on_initialize(_block_number_not_used: $trait_instance::BlockNumber) { $( $impl )* }
+			fn on_initialize(_block_number_not_used: $trait_instance::BlockNumber) {
+				#[cfg(feature = "std")]
```

Comment:
There is a nice `if_std!` macro in `sr-std` for that.

Reply:
Cool, thanks for letting me know about that stuff. It's done now

Reply:
I suppose it is fine for now, but in the future, maybe we'd want to actually make this part in dispatch.rs be gated with a special feature? so that not all nodes (as many of them, even when running the native compile) spend time on this and report and only those that have been compiled with a special flag do. I assume, for instance, as a validator behind a sentry node you don't want this stuff, nor do you want to report anything to telementry.

Reply:
I looked into this at the start and I'd say this isn't necessary - in the runtime we're calling these spans maximum a few hundred times per second, and with no targets enabled there will be (for practical purposes) no real cost to it in this situation. Here's a link to an expanded version of `balances`  https://gist.github.com/mattrutherford/8b0395a1e061c4b5f0fdbd39aea70382 

Let me know what you think?